test/data/azure/main.tf

```diff
@@ -0,0 +1,171 @@
+}
+
+# Create virtual network
+resource "azurerm_virtual_network" "testterraformnetwork" {
+    name                = "testVnet"
+    address_space       = ["10.0.0.0/16"]
+    location            = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name = data.azurerm_resource_group.testResourceGroup.name
+
+}
+
+# Create subnet
+resource "azurerm_subnet" "testterraformsubnet" {
+    name                 = "testSubnet"
+    resource_group_name  = data.azurerm_resource_group.testResourceGroup.name
+    virtual_network_name = azurerm_virtual_network.testterraformnetwork.name
+    address_prefixes       = ["10.0.1.0/24"]
+}
+
+# Create public IPs
+resource "azurerm_public_ip" "testterraformpublicip" {
+    name                         = "testPublicIP"
+    location                     = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name          = data.azurerm_resource_group.testResourceGroup.name
+    allocation_method            = "Dynamic"
+
+}
+
+# Create Network Security Group and rule
+resource "azurerm_network_security_group" "testterraformnsg" {
+    name                = "testNetworkSecurityGroup"
+    location            = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name = data.azurerm_resource_group.testResourceGroup.name
+
+    security_rule {
+        name                       = "SSH"
+        priority                   = 1001
+        direction                  = "Inbound"
+        access                     = "Allow"
+        protocol                   = "Tcp"
+        source_port_range          = "*"
+        destination_port_range     = "22"
+        source_address_prefix      = "*"
+        destination_address_prefix = "*"
+    }
+
+}
+
+# Create network interface
+resource "azurerm_network_interface" "testterraformnic" {
+    name                      = "testNIC"
+    location                  = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name       = data.azurerm_resource_group.testResourceGroup.name
+
+    ip_configuration {
+        name                          = "testNicConfiguration"
+        subnet_id                     = azurerm_subnet.testterraformsubnet.id
+        private_ip_address_allocation = "Dynamic"
+        public_ip_address_id          = azurerm_public_ip.testterraformpublicip.id
+    }
+
+}
+
+# Connect the security group to the network interface
+resource "azurerm_network_interface_security_group_association" "test" {
+    network_interface_id      = azurerm_network_interface.testterraformnic.id
+    network_security_group_id = azurerm_network_security_group.testterraformnsg.id
+}
+
+# Create (and display) an SSH key
+resource "tls_private_key" "test_ssh" {
+  algorithm = "RSA"
+  rsa_bits = 4096
+}
+output "tls_private_key" {
+    value = tls_private_key.test_ssh.private_key_pem
+    sensitive = true
+}
+
+# Create virtual machine
+resource "azurerm_linux_virtual_machine" "testterraformvm" {
```

Comment:
Can we also try `priority = "Spot"` to get a spot instance and save some more money?

Reply:
Done, I hope there won't be any problems with it, but we'll see in time.

Reply:
FTR does `priority = "Spot"` work for B series instances or does it cause a failure ? If these are connected and cause failures when one of the values doesn't match the other we at least need a comment around then, or visual separation with new lines + comments etc.

Reply:
I don't have the exact error it gives but during my manual testing it just gave an error with something like "Selected instance can not be found/provisioned".. The failure is shown during the time terraform tries to provision the resource, terraform doesn't know this before and believes this combination is fine and proceeds. If you try to manually provision a B-series virtual machine in Azure portal it will tell you "This size does not support Azure Spot"...